### DISCUSSION TITLE
more detailed build instructions for native lib support

### DISCUSSION THREAD

**Discussion by DOliana** [AsamK/signal-cli#534] [Q&A]
I would like to leverage your cli for a docker image (mqtt to signal - more when it's public ;-)). Right now I am stuck on 0.7.1 due to native library support of libsignal and libzkgroup for ARMv7.

I found your wiki page about the build setup, but was not able to get it running. Could you elaborate it or provide an example? If you point me in the right direction I would share the instructions for your wiki.

**Comment by bublath** [comment]
I know its not a direct answer to your problem, but I have the same issue (but managed to build). You can find the current libraries for 0.8.0 here:
https://github.com/bublath/FHEM-Signalbot/tree/main/armv7l
just replace them with a "zip -u xxx.jar libxxx.so"

**Reply by bublath** [reply]
+ there is a script in that project that creates a docker image automatically (well including another package but that should be easy to modify).

**Reply by DOliana** [reply]
Thank you very much. Could you provide the build script? I would prefer to have the library build at each image creation to have a current version.

In regard to the install script: I just looked at it and it looks good. BTW: its really clean and understandable scripting! I was just wondering: you pointed me to the library in your repo, but the script uses another one. Are you aware of that?
